components/search_results_item/search_results_item.jsx

```diff
@@ -1,78 +1,115 @@
-import TeamStore from 'stores/team_store.jsx';
 
 import Constants from 'utils/constants.jsx';
 import * as PostUtils from 'utils/post_utils.jsx';
 import * as Utils from 'utils/utils.jsx';
 
-import ProfilePicture from './profile_picture.jsx';
-import UserProfile from './user_profile.jsx';
-
-export default class SearchResultsItem extends React.Component {
+export default class SearchResultsItem extends React.PureComponent {
     static propTypes = {
+
+        /**
+        *  Data used for rendering post
+        */
         post: PropTypes.object,
+
+        /**
+        *  count used for passing down to PostFlagIcon, DotMenu and CommentIcon
+        */
         lastPostCount: PropTypes.number,
+
+        /**
+        *  user object for rendering profile_picture
+        */
         user: PropTypes.object,
+
+        /**
+        *  channel object for rendering channel name on top of result
+        */
         channel: PropTypes.object,
+
+        /**
+        *  Flag for determining result display setting
+        */
         compactDisplay: PropTypes.bool,
+
+        /**
+        *  Flag for highlighting mentions
+        */
         isMentionSearch: PropTypes.bool,
-        isFlaggedSearch: PropTypes.bool,
+
+        /**
+        *  Flag for highlighting mentions
+        */
         term: PropTypes.string,
+
+        /**
+        *  Flag for determining time format
+        */
         useMilitaryTime: PropTypes.bool.isRequired,
-        shrink: PropTypes.func,
+
+        /**
+        *  Flag for determining result flag state
+        */
         isFlagged: PropTypes.bool,
+
+        /**
+        *  Flag for determining profile busy status
+        */
         isBusy: PropTypes.bool,
+
+        /**
+        *  Data used for status in profile
+        */
         status: PropTypes.string,
-        onSelect: PropTypes.func
+
+        /**
+        *  Data used creating URl for jump to post
+        */
+        currentTeamName: PropTypes.string,
+
+        /**
+        *  Function used for shrinking LHS
+        *  on click of jump to message in expanded mode
+        */
+        shrink: PropTypes.func,
+
+        /**
+        *  Function used for selecting a post to comment
+        */
+        onSelect: PropTypes.func,
+
+        /**
+        *  Function used for closing LHS
+        */
+        actions: PropTypes.shape({
+            emitCloseRightHandSide: PropTypes.func.isRequired
+        }).isRequired
     };
 
     constructor(props) {
         super(props);
 
-        this.handleFocusRHSClick = this.handleFocusRHSClick.bind(this);
-        this.handleJumpClick = this.handleJumpClick.bind(this);
-        this.handleDropdownOpened = this.handleDropdownOpened.bind(this);
-        this.shrinkSidebar = this.shrinkSidebar.bind(this);
-
         this.state = {
-            currentTeamDisplayName: TeamStore.getCurrent().name,
             width: '',
             height: '',
             dropdownOpened: false
```

Comment:
I can't find where we used `width` and `height` in this component. Might need to remove this.

Reply:
This is needed for removing timestamp as link on mobile view as of now. Refer this PR: https://github.com/mattermost/mattermost-webapp/pull/435 I will change to use redux store and selector to update these components later.

Reply:
Cool, then no need to change it here. Thanks!